packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -0,0 +1,240 @@
+    }
+    return null;
+  }
+
+  void _paintVerticalThumb(Canvas canvas, Size size, double thumbOffset, double thumbExtent) {
+    final Offset thumbOrigin = new Offset(_getThumbX(size), thumbOffset);
+    final Size thumbSize = new Size(thickness, thumbExtent);
+    final Rect thumbRect = thumbOrigin & thumbSize;
+    if (radius == null)
+      canvas.drawRect(thumbRect, _paint);
+    else
+      canvas.drawRRect(new RRect.fromRectAndRadius(thumbRect, radius), _paint);
+  }
+
+  void _paintHorizontalThumb(Canvas canvas, Size size, double thumbOffset, double thumbExtent) {
+    final Offset thumbOrigin = new Offset(thumbOffset, size.height - thickness);
+    final Size thumbSize = new Size(thumbExtent, thickness);
+    final Rect thumbRect = thumbOrigin & thumbSize;
+    if (radius == null)
+      canvas.drawRect(thumbRect, _paint);
+    else
+      canvas.drawRRect(new RRect.fromRectAndRadius(thumbRect, radius), _paint);
+  }
+
+  void _paintThumb(
+    double before,
+    double inside,
+    double after,
+    double viewport,
+    Canvas canvas,
+    Size size,
+    void painter(Canvas canvas, Size size, double thumbOffset, double thumbExtent),
+  ) {
+    // Establish the minimum size possible.
+    double thumbExtent = math.min(viewport, minLength);
+    if (before + inside + after > 0.0) {
+      final double fractionVisible = inside / (before + inside + after);
+      thumbExtent = math.max(
+        thumbExtent,
+        viewport * fractionVisible - 2 * mainAxisMargin,
+      );
+    }
+
+    final double fractionPast = before / (before + after);
+    final double thumbOffset = (before + after > 0.0)
+        ? fractionPast * (viewport - thumbExtent - 2 * mainAxisMargin) + mainAxisMargin
+        : mainAxisMargin;
+
+    painter(canvas, size, thumbOffset, thumbExtent);
+  }
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    if (_lastAxisDirection == null || _lastMetrics == null || _opacity.value == 0.0)
+      return;
+    switch (_lastAxisDirection) {
+      case AxisDirection.down:
+        _paintThumb(_lastMetrics.extentBefore, _lastMetrics.extentInside, _lastMetrics.extentAfter, size.height, canvas, size, _paintVerticalThumb);
+        break;
+      case AxisDirection.up:
+        _paintThumb(_lastMetrics.extentAfter, _lastMetrics.extentInside, _lastMetrics.extentBefore, size.height, canvas, size, _paintVerticalThumb);
+        break;
+      case AxisDirection.right:
+        _paintThumb(_lastMetrics.extentBefore, _lastMetrics.extentInside, _lastMetrics.extentAfter, size.width, canvas, size, _paintHorizontalThumb);
+        break;
+      case AxisDirection.left:
+        _paintThumb(_lastMetrics.extentAfter, _lastMetrics.extentInside, _lastMetrics.extentBefore, size.width, canvas, size, _paintHorizontalThumb);
+        break;
+    }
+  }
+
+  @override
+  bool hitTest(Offset position) => null;
+
+  @override
+  bool shouldRepaint(ScrollbarPainter oldDelegate) => false;
```

Comment:
false is wrong here. It should be comparing every constructor argument, and they should all be immutable fields, so that when you create a new instance of this custom painter, the painting logic knows whether it's different or not.